lib/ibm_watson/language_translator_v3.rb

```diff
@@ -339,32 +379,192 @@ def get_model(model_id:)
-    # @param default_models [Boolean] If the default parameter isn't specified, the service will return all models
-    #   (default and non-default) for each language pair. To return only default models,
-    #   set this to `true`. To return only non-default models, set this to `false`. There
-    #   is exactly one default model per language pair, the IBM provided base model.
+    # @!method list_documents
+    # List documents.
+    # Lists documents that have been submitted for translation.
     # @return [IBMCloudSdkCore::DetailedResponse] A `IBMCloudSdkCore::DetailedResponse` object representing the response.
-    def list_models(source: nil, target: nil, default_models: nil)
+    def list_documents
       headers = {
       }
-      sdk_headers = Common.new.get_sdk_headers("language_translator", "V3", "list_models")
+      sdk_headers = Common.new.get_sdk_headers("language_translator", "V3", "list_documents")
       headers.merge!(sdk_headers)
 
       params = {
-        "version" => @version,
-        "source" => source,
-        "target" => target,
-        "default" => default_models
+        "version" => @version
       }
 
-      method_url = "/v3/models"
+      method_url = "/v3/documents"
+
+      response = request(
+        method: "GET",
+        url: method_url,
+        headers: headers,
+        params: params,
+        accept_json: true
+      )
+      response
+    end
+
+    ##
+    # @!method translate_document(file:, filename: nil, file_content_type: nil, model_id: nil, source: nil, target: nil, document_id: nil)
+    # Translate document.
+    # Submit a document for translation. You can submit the document contents in the
+    #   `file` parameter, or you can reference a previously submitted document by document
+    #   ID.
+    # @param file [File] The source file to translate.
+    #
+    #   [Supported file
+    #   types](https://cloud.ibm.com/docs/services/language-translator?topic=language-translator-document-translator-tutorial#supported-file-formats)
+    #
+    #   Maximum file size: **20 MB**.
+    # @param filename [String] The filename for file.
+    # @param file_content_type [String] The content type of file.
+    # @param model_id [String] The model to use for translation. `model_id` or both `source` and `target` are
+    #   required.
+    # @param source [String] Language code that specifies the language of the source document.
+    # @param target [String] Language code that specifies the target language for translation.
+    # @param document_id [String] To use a previously submitted document as the source for a new translation, enter
+    #   the `document_id` of the document.
+    # @return [IBMCloudSdkCore::DetailedResponse] A `IBMCloudSdkCore::DetailedResponse` object representing the response.
+    def translate_document(file:, filename: nil, file_content_type: nil, model_id: nil, source: nil, target: nil, document_id: nil)
```

Comment:
Is there a null check for `filename`? The generator should generate this param with the same "requiredness" as the file. I think in python it generates as optional but there is a null check. Just wanted to make sure since going from optional to required is breaking.

Reply:
Nope there is no null check for the filename.

Reply:
should I add it as handwritten change?